File: tools/make_sitemap_snippets.mjs
import { readdirSync, statSync } from "fs";
import { join } from "path";

const BASE = "https://opskrift-airfryer.dk";
const FOLDERS = [["opskrifter","/opskrifter"], ["guides","/guides"]];
const today = new Date().toISOString().slice(0,10);

function* walk(dir) {
  for (const name of readdirSync(dir)) {
    const p = join(dir, name);
    const s = statSync(p);
    if (s.isDirectory()) yield* walk(p);
    else if (name.toLowerCase().endsWith(".html")) yield p;
  }
}

function emit(pathWeb){
  const enc = encodeURI(pathWeb);
  return `  <url><loc>${BASE}${enc}</loc><lastmod>${today}</lastmod><changefreq>weekly</changefreq><priority>0.6</priority></url>`;
}

let out = [];
for (const [, webroot] of FOLDERS) {
  const local = "." + webroot;
  try {
    for (const p of walk(local)) {
      const web = p.replace(/\\/g,"").replace(/^\./,"").replace(/\\/g,"/"); // -> /opskrifter/...
      out.push(emit(web));
    }
  } catch {}
}

console.log(out.sort().join("\n"));

File: tools/make_sitemap_snippets.py
import os, urllib.parse, datetime

BASE = "https://opskrift-airfryer.dk"
FOLDERS = [("opskrifter", "/opskrifter"), ("guides", "/guides")]
today = datetime.date.today().isoformat()

def emit(path):
    url = f"{BASE}{urllib.parse.quote(path)}"
    return f'  <url><loc>{url}</loc><lastmod>{today}</lastmod><changefreq>weekly</changefreq><priority>0.6</priority></url>'

snippets = []

for label, webroot in FOLDERS:
    local = "." + webroot
    if not os.path.isdir(local):
        continue
    for root, _, files in os.walk(local):
        for f in files:
            if f.lower().endswith(".html"):
                rel = os.path.join(root, f)[1:]  # drop leading dot
                rel_web = rel.replace("\\", "/")
                snippets.append(emit(rel_web))

print("\n".join(sorted(snippets)))
